src/transaction/transaction_manager.cpp

```diff
@@ -268,4 +311,10 @@ void TransactionManager::DeallocateInsertedTupleIfVarlen(TransactionContext *txn
     }
   }
 }
+
+TransactionConstraint *TransactionManager::InstallConstraint(TransactionContext *txn, constraint_fn fn) {
+  auto iter = constraints_.EmplaceBack(txn->StartTime(), txn->TxnId().load(), fn);
```

Comment:
How are you uninstalling constraints?  I should see the action framework leveraged here to handle the removal and eventual deletion of the constraints for both aborts and commits.  As written your constraint list can never shrink.

Reply:
For commits we can't remove them just yet. There is no layer above that is checking the constraints, I'll put a todo in for that. But for aborts I'll add in the removal